### DISCUSSION TITLE
Redirecting infinite and returns 400 under multiple apache instances

### DISCUSSION THREAD

**Discussion by Suga94s** [OpenIDC/mod_auth_openidc#1031] [General]
Hello 
I have two apache instances behind proxy for OIDC. Each one works fine when handled separately but when I configure it to a load balancer together.
the authentication keeps looping and returns 400

<VirtualHost *:80>
OIDCClientID myclient-id
OIDCClientSecret XXXXXXXXXXXXXXXXXXXXXXXXXX
OIDCProviderMetadataURL https://auth.example.com/.well-known/openid-configuration
OIDCProviderAuthorizationEndpoint https://auth.example.com/oAuth2/authorize?values=L4
OIDCRedirectURI /redirect_uri
OIDCCryptoPassphrase xxxxxxxxxxxx

Can you please help/guide me on this issue

**Comment by zandbelt** [comment]
you'll need to configure a shared caching mechanism, see: https://github.com/OpenIDC/mod_auth_openidc/wiki/Caching

**Comment by NMurthy95** [comment]
Hello @zandbelt , 
I have implemented file cache storage mechanism in one apache server instance. 
Instance 1: 
**OIDCCacheType file
OIDCCacheDir /demoproject/cache**


I need to use this directory as common shared file cache storage for other instances. How we can do that ?  I wanted like below . 
**OIDCCacheType file
OIDCCacheDir Instance 1 ip/demoproject/cache**
 [ we tried with IP address but it's considering as URL and failed ] 



Also how to configure this directory in another instance oidc configuration. Could you please help us with examples ?
